### PR TITLE
feat: Upstream testing

### PR DESCRIPTION
**Pull request by snaselj**
- add upstream monitoring
- add workflow fragment to trigger testing
- bump minimum nautobot version to 1.4
- fix UPPERCASE env var comment in `tasks.py`
- fix csv export to be compatible with tests
- fix failing tests introduced with nautobot 1.5.10 and 1.5.11
- remove obsolete `CACHEOPS` from development nautobot configuration
- update Dockerfile to be compatible with upstream testing
- update poetry.lock

### CODE REVIEW

File: development/Dockerfile [hunk 1]
```diff
@@ -1,49 +1,83 @@
...
-# with the one in the poetry.lock
-RUN if [ "$NAUTOBOT_VER" = "stable" ]; \
-    then poetry add nautobot@latest; \
-    else poetry add nautobot==${NAUTOBOT_VER}; \
-    fi
+# # Uncomment the lines below if you are apt-installing any package.
+# RUN apt-get -y update && apt-get -y install \
+#     gcc \
+#     libmariadb-dev-compat \
+#     libmariadb-dev \
```
Comment (snaselj) [inline development/Dockerfile:47]: Is this necessary, or can be removed?

**Comment by snaselj** [comment]
There is an issue related to this:
https://github.com/nautobot/nautobot/pull/3177

How to fail tests:
```
export INVOKE_NAUTOBOT_DEVICE_LIFECYCLE_MGMT_NAUTOBOT_VER=1.5.10
export INVOKE_NAUTOBOT_DEVICE_LIFECYCLE_MGMT_PYTHON_VER=3.10
invoke build
invoke unittest
```

With `NAUTOBOT_VER=1.5.9` unittest works well.

**Comment by glennmatthews** [comment]
Looks like the tests are failing with latest because they are only creating two object instances, instead of the three or more objects that the generic test expects/requires. Should be a pretty straightforward fix.

File: development/Dockerfile (hunk 1 above)
Comment (snaselj) [reply development/Dockerfile]: Removed

**Comment by snaselj** [comment]
Hi package maintainers @jvanderaa @progala , this PR is ready for review now.
The PR description contains list of changes

File: development/Dockerfile [hunk 2]
```diff
@@ -1,49 +1,78 @@
...
 ENV NAUTOBOT_ROOT ${NAUTOBOT_ROOT}
 
-RUN apt-get update && apt-get install -y gcc libmariadb-dev-compat libmariadb-dev
+# Install Poetry manually via its installer script;
+# We might be using an older version of Nautobot that includes an older version of Poetry
+# and CI and local development may have a newer version of Poetry
+# Since this is only used for development and we don't ship this container, pinning Poetry back is not expressly necessary
+# We also don't need virtual environments in container
+RUN curl -sSL https://install.python-poetry.org | python3 - && \
+  poetry config virtualenvs.create false
```
Comment (jvanderaa) [inline development/Dockerfile:30]: What is the benefit of using the installer script versus just doing the install with pip?

Comment (bryanculver) [reply development/Dockerfile:30]: We are standardizing the install method and at present is the preferred way of installing Poetry.

**Review by bradh11 — APPROVED** [review]

**Pull request merged**